src/Chapter2.hs

```diff
@@ -349,7 +368,10 @@ Implement a function that returns only the first half of a given list.
 "b"
 -}
 -- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf l =
+    let len = div (length l) 2
+    in take len l
```

Comment:
Nice usage of `let-in` construction in here! Makes code even more readable, great taste 👍🏼